sdk/monitor/azure-opentelemetry-exporter-azuremonitor/src/test/java/com/azure/opentelemetry/exporter/azuremonitor/EventHubsExporterIntegrationTest.java

```diff
@@ -0,0 +1,121 @@
+        try {
+            producer.createBatch()
+                .flatMap(batch -> {
+                    batch.tryAdd(new EventData("test event "));
+                    return producer.send(batch);
+                }).subscribe();
+        } finally {
+            span.end();
+            scope.close();
+        }
+        assertTrue(exporterCountDown.await(5, TimeUnit.SECONDS));
+    }
+
+
+    @Test
+    public void processorTest() throws InterruptedException {
+        CountDownLatch exporterCountDown = new CountDownLatch(2);
+        EventHubProducerAsyncClient producer = new EventHubClientBuilder()
+            .connectionString(CONNECTION_STRING)
+            .buildAsyncProducerClient();
+
+        Tracer tracer = configureAzureMonitorExporter((context, next) -> {
+            Mono<String> asyncString = FluxUtil.collectBytesInByteBufferStream(context.getHttpRequest().getBody())
+                .map(bytes -> new String(bytes, StandardCharsets.UTF_8));
+            asyncString.subscribe(value -> {
+                if (value.contains("event-hubs-consumer-testing")) {
+                    exporterCountDown.countDown();
+                }
+                if (value.contains("EventHubs.process")) {
+                    exporterCountDown.countDown();
+                }
+            });
+            return next.process();
+        });
+
+        CountDownLatch partitionOwned = new CountDownLatch(1);
+        CountDownLatch eventCountDown = new CountDownLatch(1);
+        EventProcessorClient processorClient = new EventProcessorClientBuilder()
+            .consumerGroup(EventHubClientBuilder.DEFAULT_CONSUMER_GROUP_NAME)
+            .connectionString(CONNECTION_STRING)
+            .processPartitionInitialization(partition -> {
+                if (partition.getPartitionContext().getPartitionId().equals("0")) {
+                    partitionOwned.countDown();
+                }
+            })
+            .processEvent(event -> eventCountDown.countDown())
+            .processError(error -> { })
+            .loadBalancingStrategy(LoadBalancingStrategy.GREEDY)
+            .checkpointStore(new InMemoryCheckpointStore())
+            .buildEventProcessorClient();
+
+        Span span = tracer.spanBuilder("event-hubs-consumer-testing").startSpan();
+        final Scope scope = span.makeCurrent();
+        try {
+            processorClient.start();
+        } finally {
+            span.end();
+            scope.close();
+        }
+        partitionOwned.await(10, TimeUnit.SECONDS);
+
+        // send an event after partition 0 is owned
+        producer.createBatch(new CreateBatchOptions().setPartitionId("0"))
+            .flatMap(batch -> {
+                batch.tryAdd(new EventData("test event "));
+                return producer.send(batch);
+            }).block();
+
+        assertTrue(eventCountDown.await(10, TimeUnit.SECONDS));
+        assertTrue(exporterCountDown.await(10, TimeUnit.SECONDS));
+    }
+}
```

Comment:
Do we need a comprehensive test for this https://gist.github.com/pakrym/8902e50db6cb9ce431083ad461dae2cc#all-working-together

Reply:
Added storage checkpoint store which will cover the comprehensive test scenario.